### PR TITLE
Last property typehints

### PR DESCRIPTION
**Pull request by jack-worman**
Should be everything without breaking BC

### CODE REVIEW

**Comment by orklah** [comment]
Thanks! Could you make another sweep of performance check?

File: phpcs.xml [hunk 1]
```diff
@@ -228,17 +228,6 @@
     -->
     <rule ref="SlevomatCodingStandard.PHP.ShortList"/>
 
-    <!--
-        Requires arrow functions.
-        https://github.com/slevomat/coding-standard/#slevomatcodingstandardfunctionsrequirearrowfunction-
-    -->
-    <rule ref="SlevomatCodingStandard.Functions.RequireArrowFunction">
```
Comment (weirdan) [inline phpcs.xml:235]: Is there any reason why this got dropped?

File: src/Psalm/Config.php [hunk 2]
```diff
@@ -559,10 +559,7 @@ class Config
      */
     public $debug_emitted_issues = false;
 
-    /**
-     * @var bool
-     */
-    private $report_info = true;
+    private bool $report_info = true;
```
Comment (weirdan) [inline src/Psalm/Config.php:562]: Not sure how this differs from, say, `$max_string_length`. Why did one get promoted to native types and another did not?

Comment (jack-worman) [reply src/Psalm/Config.php:562]: `$max_string_length` is public and would therefore be a BC break.

File: phpcs.xml (hunk 1 above)
Comment (jack-worman) [reply phpcs.xml:235]: After changing the phpcs PHP version to 70400 this rule started to activate. I thought it produced messier code. The writer should be able to determine whether to use a short closure or traditional closure.

File: src/Psalm/Config.php (hunk 2 above)
Comment (weirdan) [reply src/Psalm/Config.php:562]: Ok, makes sense.

File: phpcs.xml (hunk 1 above)
Comment (weirdan) [reply phpcs.xml:235]: For a lot of people it's a matter of habit (arrow functions appeared just a couple of years ago), so I'd keep it as a nudge to use arrow functions.

**Review by weirdan — APPROVED** [review]
LGTM!

**Comment by jack-worman** [comment]
> Thanks! Could you make another sweep of performance check?

```
OLD (v5.2.0):
230.14s user 20.82s system 285% cpu 1:27.77 total
228.89s user 20.20s system 295% cpu 1:24.40 total
232.80s user 19.65s system 294% cpu 1:25.61 total
232.18s user 20.11s system 296% cpu 1:25.19 total
233.24s user 19.62s system 297% cpu 1:24.93 total
235.84s user 19.35s system 296% cpu 1:26.10 total

NEW (this branch):
226.39s user 20.13s system 296% cpu 1:23.03 total
239.85s user 22.12s system 284% cpu 1:32.00 total
236.67s user 20.63s system 291% cpu 1:28.34 total
243.50s user 21.69s system 286% cpu 1:32.49 total
239.61s user 20.60s system 295% cpu 1:28.07 total
224.56s user 19.75s system 289% cpu 1:24.50 total
213.04s user 17.40s system 302% cpu 1:16.12 total
```
Again, basically the same aside.

**Comment by jack-worman** [comment]
Another test, but this time I closed out of other programs and didn't stream a twitch,
```txt
OLD:
real    1m18.422s user    3m38.936s sys     0m17.202s
real    1m15.243s user    3m33.943s sys     0m16.369s
real    1m14.178s user    3m25.938s sys     0m15.956s
real    1m13.769s user    3m25.643s sys     0m15.819s
real    1m19.965s user    3m37.302s sys     0m16.893s
real    1m12.128s user    3m23.238s sys     0m15.830s

NEW:                                                
real    1m12.951s user    3m26.481s sys     0m15.885s
real    1m12.857s user    3m25.852s sys     0m15.832s
real    1m12.813s user    3m25.501s sys     0m15.834s
real    1m20.767s user    3m37.414s sys     0m17.310s
real    1m20.317s user    3m43.043s sys     0m21.136s
real    1m13.754s user    3m26.978s sys     0m16.150s
```

**Comment by orklah** [comment]
Thanks!

**Pull request merged**